internal/provider/data_certificate.go

```diff
@@ -0,0 +1,298 @@
+						},
+						"public_key": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The public key of the certificate.",
+						},
+						"public_key_algorithm": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The algorithm used for the public key.",
+						},
+						"serial_number": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The serial number of the certificate.",
+						},
+						"signature_algorithm": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The algorithm used for the certificate's signature.",
+						},
+						"subject": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The subject of the certificate.",
+						},
+						"subject_cn": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The subject_cn(common name) of the certificate.",
+						},
+						"subject_ou": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The subject_ou(org unit) of the certificate.",
+						},
+						"subject_org": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The subject_org of the certificate.",
+						},
+						"subject_locality": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The subject_locality of the certificate.",
+						},
+						"subject_st": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The subject_st(state) of the certificate.",
+						},
+						"subject_country": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The subject_country of the certificate.",
+						},
+						"subject_alternative_name": {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "The subject alternative names in the certificate.",
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+						"thumbprint": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The thumbprint of the certificate.",
+						},
+						"thumbprint_algorithm": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The algorithm used to generate the thumbprint.",
+						},
+						"version": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The version of certificate.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataCertificateRead(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(*api_client.SddcManagerClient).ApiClient
+	log.Print("[DEBUG] Function dataCertificateRead start")
+	// Extract the domain_id from ResourceData
+	domainId, ok := data.Get("domain_id").(string)
```

Comment:
You should add a validation function in the schema instead.
It would make it impossible for the `domain_id` to be a non-string or an otherwise invalid value (e.g. empty)

Reply:
Good point. As the schema has a type parameter to validate the type of domain_id and resource_fqdn so  I changed to use the StringIsNotEmpty function from github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation to validate the inputs are not an empty string.